Jetcaster/app/src/main/java/com/example/jetcaster/ui/podcast/PodcastDetailsScreen.kt

```diff
@@ -0,0 +1,336 @@
+        modifier = modifier,
+    )
+}
+
+@Composable
+fun PodcastDetailsScreen(
+    podcast: PodcastInfo,
+    episodes: List<EpisodeInfo>,
+    toggleSubscribe: (PodcastInfo) -> Unit,
+    onQueueEpisode: (PlayerEpisode) -> Unit,
+    navigateToPlayer: (EpisodeInfo) -> Unit,
+    navigateBack: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val coroutineScope = rememberCoroutineScope()
+    val snackbarHostState = remember { SnackbarHostState() }
+    val snackBarText = stringResource(id = R.string.episode_added_to_your_queue)
+    Scaffold(
+        modifier = modifier.fillMaxSize(),
+        topBar = {
+            PodcastDetailsTopAppBar(
+                navigateBack = navigateBack,
+                modifier = Modifier.fillMaxWidth()
+            )
+        },
+        snackbarHost = {
+            SnackbarHost(hostState = snackbarHostState)
+        }
+    ) { contentPadding ->
+        PodcastDetailsContent(
+            podcast = podcast,
+            episodes = episodes,
+            toggleSubscribe = toggleSubscribe,
+            onQueueEpisode = {
+                coroutineScope.launch {
+                    snackbarHostState.showSnackbar(snackBarText)
+                }
+                onQueueEpisode(it)
+            },
+            navigateToPlayer = navigateToPlayer,
+            modifier = Modifier.padding(contentPadding)
+        )
+    }
+}
+
+@Composable
+fun PodcastDetailsContent(
+    podcast: PodcastInfo,
+    episodes: List<EpisodeInfo>,
+    toggleSubscribe: (PodcastInfo) -> Unit,
+    onQueueEpisode: (PlayerEpisode) -> Unit,
+    navigateToPlayer: (EpisodeInfo) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    LazyColumn(
+        modifier.fillMaxSize()
+    ) {
+        item {
+            PodcastDetailsHeaderItem(
+                podcast = podcast,
+                toggleSubscribe = toggleSubscribe,
+                modifier = Modifier.fillMaxWidth()
+            )
+        }
+        items(episodes, key = { it.uri }) { episode ->
+            EpisodeListItem(
+                episode = episode,
+                podcast = podcast,
+                onClick = { navigateToPlayer(episode) },
+                onQueueEpisode = onQueueEpisode,
+                modifier = Modifier.fillMaxWidth(),
+                showPodcastImage = false
+            )
+        }
+    }
+}
+
+@Composable
+fun PodcastDetailsHeaderItem(
+    podcast: PodcastInfo,
+    toggleSubscribe: (PodcastInfo) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Column(
+        modifier = modifier.padding(Keyline1)
+    ) {
+        Row(
+            verticalAlignment = Alignment.Bottom,
+            modifier = Modifier.fillMaxWidth()
+        ) {
+            AsyncImage(
+                model = ImageRequest.Builder(LocalContext.current)
+                    .data(podcast.imageUrl)
+                    .crossfade(true)
+                    .build(),
+                contentDescription = stringResource(id = R.string.cd_podcast_image),
```

Comment:
This isn't useful as a content description. If one isn't available from the model, then I'd prefer null here so that screen readers can use the title below instead of repeating a generic image description.

Reply:
Thanks for the a11y call out. `null` makes sense here since it's a decorative element.